src/evm-block-extractor/src/database/postgres_db_client.rs

```diff
@@ -97,35 +103,43 @@ impl DatabaseClient for PostgresDbClient {
 
         let mut tx = self.pool.begin().await?;
 
-        for block in blocks {
-            let block_id = block.number.0.as_u64();
-
-            sqlx::query("INSERT INTO EVM_BLOCK (id, data) VALUES ($1, $2)")
-                .bind(block_id as i64)
-                .bind(serde_json::to_value(block)?)
-                .execute(&mut *tx)
-                .await
-                .map_err(|e| anyhow::anyhow!("Error inserting block {}: {:?}", block_id, e))
-                .map(|_| ())?;
+        if !blocks.is_empty() {
```

Comment:
Please add tests for:
- saving blocks with no TXs and no Receipts
- saving blocks with TXs and no Receipts
- saving blocks with no TXs and Receipts
- saving TXs and Receipts with no Blocks (this one is strange but the client allows it)

Reply:
Well, the tests for all of these will pass, because the emulator allows this, I don’t know what is up with google big query.